bridge-ext/src/main/java/com/liferay/faces/bridge/ext/filter/internal/LiferayURLGeneratorBaseImpl.java

```diff
@@ -125,31 +129,44 @@
 	 * @param  baseURL            The String-based URL generated by Liferay Portal's PortletURLImpl#toString() method.
 	 * @param  responseNamespace  The response namespace.
 	 */
-	public LiferayURLGeneratorBaseImpl(String baseURL, String responseNamespace) {
-		this(baseURL, null, responseNamespace, null);
+	public LiferayURLGeneratorBaseImpl(String baseURL, String responseNamespace, String encoding) {
+		this(baseURL, responseNamespace, null, null, encoding);
 	}
 
 	/**
 	 * Constructs a new instance.
 	 *
-	 * @param  baseURL            The String-based URL generated by Liferay Portal's
+	 * @param  portletURL         The String-based URL generated by Liferay Portal's
 	 *                            com.liferay.portlet.PortletURLImpl#toString() method. method.
-	 * @param  portletMode        The portlet mode specified by Liferay Portal's
-	 *                            com.liferay.portlet.PortletURLImpl#getPortletMode() method. method.
 	 * @param  responseNamespace  The response namespace.
-	 * @param  windowState        The window state specified by Liferay Portal's
-	 *                            com.liferay.portlet.PortletURLImpl#getWindowState() method. method.
 	 */
-	public LiferayURLGeneratorBaseImpl(String baseURL, PortletMode portletMode, String responseNamespace,
-		WindowState windowState) {
+	public LiferayURLGeneratorBaseImpl(PortletURL portletURL, String responseNamespace, String encoding) {
+		this(portletURL.toString(), responseNamespace, portletURL.getPortletMode(), portletURL.getWindowState(),
+			encoding);
+	}
+
+	private LiferayURLGeneratorBaseImpl(String baseURL, String responseNamespace, PortletMode portletMode,
+		WindowState windowState, String encoding) {
 
 		this.baseURL = baseURL;
-		this.initialPortletMode = portletMode;
 		this.responseNamespace = responseNamespace;
+		this.initialPortletMode = portletMode;
 		this.initialWindowState = windowState;
+		this.encoding = encoding;
 		parse();
 	}
 
+	public static String encodeParameterNameOrValue(String name, String encoding) throws UnsupportedEncodingException {
```

Comment:
@stiemannkj1 Maybe the parameter should be named `nameOrValue` instead of `name`?

Reply:
@ngriffin7a, sure will do in follow up PR for both bridge-impl and bridge-ext.